commerce-payment-service/src/main/java/com/liferay/commerce/payment/service/impl/CommercePaymentMethodGroupRelLocalServiceImpl.java

```diff
@@ -306,6 +322,78 @@ public CommercePaymentMethodGroupRel setActive(
 
 		commercePaymentMethodGroupRel.setActive(active);
 
+		if (!active) {
+			Map<String, CommercePaymentMethod> commercePaymentMethodsMap =
+				_commercePaymentMethodRegistry.getCommercePaymentMethods();
+
+			List<CommercePaymentMethod> recurringCommercePaymentMethods =
+				new ArrayList<>();
+
+			for (CommercePaymentMethod commercePaymentMethod :
+					commercePaymentMethodsMap.values()) {
+
+				if (commercePaymentMethod.isProcessRecurringEnabled()) {
+					recurringCommercePaymentMethods.add(commercePaymentMethod);
+				}
+			}
+
+			if (recurringCommercePaymentMethods.size() == 1) {
+				CommercePaymentMethod lastRecurringCommercePaymentMethod =
+					recurringCommercePaymentMethods.get(0);
+
+				String lastRecurringCommercePaymentMethodKey =
+					lastRecurringCommercePaymentMethod.getKey();
+
+				if (!lastRecurringCommercePaymentMethodKey.equals(
+						commercePaymentMethodGroupRel.getEngineKey())) {
+
+					return commercePaymentMethodGroupRelPersistence.update(
+						commercePaymentMethodGroupRel);
+				}
+
+				SearchContext searchContext = new SearchContext();
+
+				Map<String, Serializable> attributes = new HashMap<>();
+
+				attributes.put(Field.STATUS, WorkflowConstants.STATUS_APPROVED);
+
+				attributes.put(CPDefinitionIndexer.FIELD_PUBLISHED, true);
+				attributes.put(
+					CPDefinitionIndexer.FIELD_SUBSCRIPTION_ENABLED, true);
+
+				long groupId = commercePaymentMethodGroupRel.getGroupId();
+
+				attributes.put("commerceChannelGroupId", groupId);
+
+				searchContext.setAttributes(attributes);
+
+				searchContext.setCompanyId(
+					commercePaymentMethodGroupRel.getCompanyId());
+
+				CPDataSourceResult cpDataSourceResult =
+					_cpDefinitionHelper.search(
+						groupId, searchContext, new CPQuery(),
+						QueryUtil.ALL_POS, QueryUtil.ALL_POS);
+
+				List<CPCatalogEntry> cpCatalogEntries =
+					cpDataSourceResult.getCPCatalogEntries();
+
+				for (CPCatalogEntry cpCatalogEntry : cpCatalogEntries) {
+					CPDefinition cpDefinition =
+						_cpDefinitionLocalService.fetchCPDefinition(
+							cpCatalogEntry.getCPDefinitionId());
+
+					if (cpDefinition != null) {
+						cpDefinition.setSubscriptionEnabled(false);
+						cpDefinition.setStatus(WorkflowConstants.STATUS_DRAFT);
+
+						_cpDefinitionLocalService.updateCPDefinition(
+							cpDefinition);
+					}
+				}
```

Comment:
@alecsloan can you put all this into a private method, maybe something like `_checkRecurringCommercePaymentMethods`? As-is, it feels really weird to have such a huge block inside a method named `setActive` -- which seems like it'd be simple -- that isn't really setting "active" at all, just taking care of a side effect. I think putting it into a private method that's called if the conditions are right (`!active`) will make a lot more sense in the future.